rust/src/x509/mod.rs

```diff
@@ -85,13 +85,17 @@ pub unsafe extern "C" fn SCX509Decode(
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn SCX509GetSubject(ptr: *const X509) -> *mut c_char {
+pub unsafe extern "C" fn SCX509GetSubject(ptr: *const X509, subject_name: *mut *mut u8, subject_len: *mut u32) {
     if ptr.is_null() {
-        return std::ptr::null_mut();
+        *subject_len = 0;
+        *subject_name = std::ptr::null_mut();
+        return;
     }
     let x509 = cast_pointer! {ptr, X509};
-    let subject = x509.0.tbs_certificate.subject.to_string();
-    rust_string_to_c(subject)
+    let subject = x509.0.tbs_certificate.subject().to_string().into_bytes();
```

Comment:
Should we still do `to_string` ? I am not sure which code runs.
Should we not do `as_ref` ? https://github.com/rusticata/x509-parser/blob/b7dcc9397b596cf9fa3df65115c3f405f1748b2a/src/certificate.rs#L627C8-L627C14

Reply:
I'm a bit unclear here. `AsRef` trait (linked) seems to be defined for `TbsCertificate` but we're dealing with `X509Name` once we extract the subject from the certificate.
Could you please tell what should I change?

Reply:
How would `to_string` deal with null chars and the like?